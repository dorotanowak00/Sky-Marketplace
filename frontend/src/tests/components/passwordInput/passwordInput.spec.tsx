import { fireEvent, render, screen } from "@testing-library/react";
import { PasswordInput, RenderInputProps } from "components/passwordInput/PasswordInput";
import { SyntheticEvent } from "react";

const inputChangeHander = jest.fn();
const value = "";

const passwordInputProps: RenderInputProps = {
    item: {
        id: "passwordInput",
        label: "Item Label",
        placeholder: "Password Input",
    },
    value: value,
    onInputChange: inputChangeHander,
};

const passwordInputTestId = "passwordInput";

describe("components/passwordInput", () => {
    it("renders the input", () => {
        const { container } = render(
            <PasswordInput
                item={passwordInputProps.item}
                value={passwordInputProps.value}
                onInputChange={passwordInputProps.onInputChange}
            />
        );
        expect(container).toMatchSnapshot();
    });

    it("updates model after change", () => {
        inputChangeHander.mockImplementationOnce((evt: SyntheticEvent) => {
            expect((evt.target as HTMLInputElement).value).toBe("new value");
        });

        render(
            <PasswordInput
                item={passwordInputProps.item}
                value={passwordInputProps.value}
                onInputChange={passwordInputProps.onInputChange}
            />
        );
        fireEvent.change(screen.getByTestId(passwordInputTestId), {
            target: { value: "new value" },
        });
    });

    it("sets the width inline styles when passed", () => {
        const { container } = render(
            <PasswordInput
                item={passwordInputProps.item}
                value={passwordInputProps.value}
                onInputChange={passwordInputProps.onInputChange}
                width={"100px"}
            />
        );
        expect(container).toMatchSnapshot();
    });
});
