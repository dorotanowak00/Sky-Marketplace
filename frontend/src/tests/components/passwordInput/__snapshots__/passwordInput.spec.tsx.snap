// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/passwordInput renders the input 1`] = `
<div>
  <div
    class="inputContainer"
  >
    <label
      class="label"
      for="passwordInput"
    >
      Item Label
    </label>
    <input
      class="input"
      data-testid="passwordInput"
      id="passwordInput"
      placeholder="Password Input"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`components/passwordInput sets the width inline styles when passed 1`] = `
<div>
  <div
    class="inputContainer"
    style="width: 100px;"
  >
    <label
      class="label"
      for="passwordInput"
    >
      Item Label
    </label>
    <input
      class="input"
      data-testid="passwordInput"
      id="passwordInput"
      placeholder="Password Input"
      type="text"
      value=""
    />
  </div>
</div>
`;
